examples/Web/embed_external_rois.html

```diff
@@ -0,0 +1,426 @@
+                var theZ = viewport.getZPos();
+                var filter = viewport.viewportimg.get(0).get_current_rois_filter();
+                viewport.viewportimg.get(0).refresh_rois(theZ, theT, filter);
+            }
+        };
+
+        var hide_rois = function () {
+            if (viewport.viewportimg.get(0).hide_rois) {
+                viewport.viewportimg.get(0).hide_rois(true, false);
+            }
+        }
+
+        var _imageLoad = function (ev, viewport) {
+            $('#image-name').html(viewport.loadedImg.meta.imageName);
+
+            var tmp = viewport.getPixelSizes();
+            if (tmp.x !== 0) {
+                $("#viewport-scalebar").prop("disabled", false);
+                $("#viewport-scalebar").prop("checked", true);
+                show_scalebar();
+            }
+
+            $("#viewport-show-rois").click(function () {
+                show_rois();
+            });
+
+            $("#viewport-hide-rois").click(function () {
+                hide_rois();
+            });
+
+            $("#viewport-add-rectangle").click(function () {
+                add_random_rectangle();
+            });
+
+            $("#viewport-add-ellipse").click(function () {
+                add_random_ellipse();
+            });
+
+            $("#viewport-add-line").click(function () {
+                add_random_line();
+            });
+
+            $("#pop-roi").click(function () {
+                pop_roi();
+            });
+
+            $("#pull-roi").click(function () {
+                pull_roi();
+            });
+
+            $("#flush-colors").click(function() {
+                flush_colors();
+                var $reset = $("#reset-shapes");
+                if ($reset.is(":disabled")) {
+                    $reset.removeAttr("disabled");
+                }
+            });
+
+            $("#reset-shapes").click(function() {
+                viewport.viewportimg.get(0).restore_shapes();
+                $("#reset-shapes").attr("disabled", "disabled");
+            })
+
+            $("#viewport-scalebar").change(function () {
+                if (this.checked) {
+                    show_scalebar();
+                } else {
+                    hide_scalebar();
+                }
+            });
+        };
+
+        var instant_zoom = function (e, percent) {
+            if (viewport.viewportimg.get(0).setRoiZoom) {
+                viewport.viewportimg.get(0).setRoiZoom(percent);
+            }
+            if (viewport.viewportimg.get(0).setScalebarZoom) {
+                viewport.viewportimg.get(0).setScalebarZoom(percent / 100);
+            }
+        };
+
+        $(document).ready(function () {
+            viewport = $.WeblitzViewport($("#viewport"), "http://OMERO_WEB_HOST/webgateway/", {
+                'mediaroot': "http://OMERO_WEB_HOST/static/"
+            });
+
+            viewport.bind('imageLoad', _imageLoad);
+            viewport.bind('instant_zoom', instant_zoom);
```

Comment:
I also had to add 
```
viewport.bind('imageChange', refresh_rois);
```
here so that the shapes are refreshed when the viewer changes Z / T position.